Mathlib/Algebra/Group/Prod.lean

```diff
@@ -447,43 +447,22 @@ variable {M N P : Type*} [Mul M] [Mul N] [Semigroup P]
 /-- Coproduct of two `MulHom`s with the same codomain with commutation assumption :
   `f.noncommCoprod g _ (p : M × N) = f p.1 * g p.2`.
   (For the commutative case, use `MulHom.coprod`) -/
-@[to_additive
+@[to_additive (attr := simps)
     "Coproduct of two `AddHom`s with the same codomain with commutation assumption:
     `f.noncommCoprod g _ (p : M × N) = f p.1 + g p.2`.
     (For the commutative case, use `AddHom.coprod`)"]
 def noncommCoprod : M × N →ₙ* P where
-  toFun mn := (f mn.fst) * (g mn.snd)
-  map_mul' mn mn' := by
-    simp only [Prod.fst_mul, Prod.snd_mul, map_mul]
-    simp only [mul_assoc]
-    apply congr_arg₂ _ rfl
-    simp only [← mul_assoc]
-    apply congr_arg₂ _ _ rfl
-    apply comm
-
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply (p : M × N) :
-  f.noncommCoprod g comm p = f p.1 * g p.2 :=
-  rfl
-
-/- -- simp_normal_form complains
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply' (p : M × N) :
-    f.noncommCoprod g comm p = g p.2 * f p.1 :=
-  comm p.1 p.2
--/
+  toFun mn := f mn.fst * g mn.snd
+  map_mul' mn mn' := by simpa using (comm _ _).mul_mul_mul_comm _ _
 
 @[to_additive]
 theorem comp_noncommCoprod {Q : Type*} [Semigroup Q] (h : P →ₙ* Q) :
     h.comp (f.noncommCoprod g comm) =
-      (h.comp f).noncommCoprod (h.comp g) (fun m n ↦ by
-        simp only [coe_comp, Function.comp_apply]
-        exact Commute.map (comm m n) h) :=
```

Comment:
The trick here is to note that the `simp only` line looks like obvious definitional equalities, so can be removed.